tests/foreman/api/test_contentviewversion.py

```diff
@@ -129,3 +133,64 @@ def test_delete_version_negative(self):
             content_view.version[0].delete()
         # Make sure that content view version is still present
         self.assertEqual(len(content_view.read().version), 1)
+
+    def test_incremental_update_puppet(self):
+        """@Test: Incrementally update a CVV with a puppet module.
+
+        @Assert: The the incremental update succeeds with no errors, and the
+        content view is given an additional version.
+
+        @Feature: ContentViewVersion
+
+        """
+        # Create a puppet repository and upload a puppet module into it.
+        product = entities.Product().create()
+        repository = entities.Repository(
+            content_type='puppet',
+            product=product,
+        ).create()
+        with open(get_data_file(PUPPET_MODULE_NTP_PUPPETLABS), 'rb') as handle:
+            repository.upload_content(files={'content': handle})
+
+        # Create a content view version.
+        content_view = entities.ContentView(
+            organization=product.organization,
+        ).create()
+        content_view.publish()
```

Comment:
Yes you can add another puppet module.  Also you can have any repo - yum, puppet or docker to publish a legal content view. Then you can perform incremental update.